[CORL-1700] Live Chat: reply notifications + reply tweaks

nick-funk opened the pull request:
## What does this PR do?

- Highlight replies to the viewer's comments in the live chat
- Swap "reply" for "view conversation" on the reply comments in the main live chat stream
- When a user selects a reply comment and clicks "view conversation" the conversation thread opens and does the following:
   - jumps to the reply comment they clicked on to open the conversation
   - highlights the reply comment so they have context and focus based on the thread reply they clicked on

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Go to live chat
- Create a comment (original commenter)
- Use another account (replier) to reply to that comment
- See that the reply is highlighted with an emphasized dark grey L-bracket when it comes in on the original commenter account
- Create some more replies to the original comment
- Click on the replies new "view conversation" button and see that the conversation view shows the comment you came from as well as highlights it

Some screenies:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/5751504/116126323-bd318d80-a683-11eb-890a-061faf53c819.png">

<img width="479" alt="image" src="https://user-images.githubusercontent.com/5751504/116126350-c9b5e600-a683-11eb-92d4-b02ab5500ae9.png">

nick-funk committed:
Highlight replies to viewer's comments

nick-funk committed:
Show "read conversation" instead of "reply" on live chat replies

nick-funk committed:
Highlight the selected reply comment when in conversation view

nick-funk committed:
Jump to selected reply when opening conversation view

alinhle commented on src/core/client/stream/tabs/Live/LiveComment/LiveCommentContainer.tsx:
```diff
@@ -84,6 +85,12 @@ const LiveCommentContainer: FunctionComponent<Props> = ({
+    viewer &&
+    comment.parent &&
+    comment.parent.author &&
+    comment.parent.author.id === viewer.id;
```
Could be replaced with:
`const isReplyToViewer = comment.parent?.author?.id === viewer?.id;`

alinhle commented on src/core/client/stream/tabs/Live/LiveChatContainer.tsx:
```diff
@@ -304,7 +309,14 @@ const LiveChatContainer: FunctionComponent<Props> = ({
+        id: comment.id,
+        cursor: comment.createdAt,
+      });
       showConversation(parent, "replyToParent");
```
It would be great to refactor all of the conversation stuff into a hook:
```ts
const [showConversation, hideConversation, conversationState] = useConversation();
showConversation(comment, { highlight: commentToHighlight });
```

alinhle commented on src/core/client/stream/tabs/Live/LiveComment/LiveCommentActionsContainer.tsx:
```diff
@@ -188,7 +188,27 @@ const LiveCommentActionsContainer: FunctionComponent<Props> = ({
           className={styles.action}
         />
-        {onReply && (
+        {comment.parent && onReply && (
```
Line 226 has the same implementation of the `button` with just a different handler. It should be possible to combine the two.

alinhle commented on src/core/client/stream/tabs/Live/LiveConversation/LiveConversationQuery.tsx:
```diff
@@ -36,9 +43,16 @@ const LiveConversationQuery: FunctionComponent<Props> = ({
+  const [cursor, setCursor] = useState(
+    highlightedCommentState
+      ? new Date(
+          new Date(highlightedCommentState.cursor).getTime() - 10
```
Don't substract by `- 10` it's a hack. Instead fix what's wrong with the query. In the Convo View the Before Container should always set `inclusive: false` and the After Container to `inclusive: true`. At the moment this is flipped around which is wrong.

alinhle commented on src/core/client/stream/tabs/Live/LiveComment/LiveCommentContainer.css:
```diff
@@ -36,6 +59,16 @@
+  font-size: 20px;
+  width: 20px;
+  margin-left: 16px;
+  margin-right: 12px;
```
Should use `var(--spacing-x)`

alinhle commented on src/core/client/stream/tabs/Live/LiveComment/LiveCommentContainer.css:
```diff
@@ -44,6 +77,14 @@
+  color: var(--palette-text-000);
+
+  margin-right: var(--spacing-2);
+  height: 20px;
```
Don't use `height: 20px` here on this typography element. Did you mean `line-height` instead?

alinhle commented on src/core/client/stream/tabs/Live/LiveComment/LiveCommentContainer.css:
```diff
@@ -57,6 +98,19 @@
+  color: var(--palette-text-000);
+
+  margin-right: var(--spacing-1);
+  height: 20px;
```
Don't use `height: 20px` here on this typography element. Did you mean `line-height` instead?

alinhle reviewed (changes requested):
When the highlighted comment ends up in the `beforeComments` due to cursor changes like `JumpToX`, the styling is incorrect:
![image](https://user-images.githubusercontent.com/14221600/116431230-e26ef900-a847-11eb-8550-ef6034a2a07a.png)

Good first draft.

nick-funk committed:
Use conditional operator

nick-funk committed:
Refactor reply buttons to remove duplication of button logic

nick-funk committed:
Clean up notification reply alignment CSS

nick-funk committed:
Remove conv view cursor hack

Use the inclusives on the before/after queries to make sure
that this always loads appropriately

nick-funk committed:
Add missing highlight flag in conversation "before" replies

nick-funk commented:
> When the highlighted comment ends up in the `beforeComments` due to cursor changes like `JumpToX`, the styling is incorrect:
> ![image](https://user-images.githubusercontent.com/14221600/116431230-e26ef900-a847-11eb-8550-ef6034a2a07a.png)
> 
> Good first draft.

Fixed, was just me forgetting to set the highlight flag on the `LiveReplyContainer`

nick-funk committed:
Convert conversation state management into useConversation hook

alinhle committed:
chore: refactor ReadConverstaionButton

alinhle reviewed (changes requested):
According to design, there is no whitespace/padding on the left of the comment when highlighted.
![image](https://user-images.githubusercontent.com/14221600/116472595-5fb16280-a876-11eb-906b-466e6f1fb14c.png)

nick-funk committed:
Fix padding/alignment of emphasized replies in live chat stream

nick-funk commented:
> According to design, there is no whitespace/padding on the left of the comment when highlighted.
> ![image](https://user-images.githubusercontent.com/14221600/116472595-5fb16280-a876-11eb-906b-466e6f1fb14c.png)

<img width="387" alt="image" src="https://user-images.githubusercontent.com/5751504/116478752-d5e4a380-a83b-11eb-802d-635ec33affdb.png">

Fixed

alinhle committed:
fix+chore: cleanup and fix styles

alinhle committed:
fix: edit highlighting in conversation view

alinhle committed:
fix: editing background in live chat

alinhle committed:
fix: layout actions buttons
